### PR TITLE
riff: init at 1.0.0

### PR DESCRIPTION
**Pull request by lucperkins**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
should we disable telemetry by default?

btw it has a hash mismatch on linux, not sure if it works on darwin

**Comment by lucperkins** [comment]
@figsoda Telemetry isn't enabled/disabled during build but rather on a per-command basis in the tool itself.

Which hash is currently mismatched on Linux?

**Comment by alyssais** [comment]
> @figsoda Telemetry isn't enabled/disabled during build but rather on a per-command basis in the tool itself.

We could still disable it by default with a patch or wrapper, and I think it would be worth doing so.  As a distro we have a responsibility to our users, and making telemetry opt-in for them where we can is a nice thing to do for them.

**Comment by alyssais** [comment]
> Which hash is currently mismatched on Linux?

See the OfBorg failures, e.g. https://github.com/NixOS/nixpkgs/pull/190026/checks?check_run_id=8211840630.

File: pkgs/development/tools/riff/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "DeterminateSystems";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-q1cI7G/c7kVwMFMjbmdbNdrU6GHVmxTxcLbAfLaMEYU=";
+  };
+
+  cargoSha256 = "sha256-vmv7SOQqAPzYgKoTJKdiRrJ5t8iU1QlZy+VSlv8FRQE";
+
+  nativeBuildInputs = [ pkg-config ];
```
Comment (alyssais) [inline pkgs/development/tools/riff/default.nix:23]:
```suggestion
  nativeBuildInputs = [ makeWrapper pkg-config ];
```

File: pkgs/development/tools/riff/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, openssl
+, pkg-config
```
Comment (alyssais) [inline pkgs/development/tools/riff/default.nix:5]:
```suggestion
, makeWrapper
, pkg-config
```

File: pkgs/development/tools/riff/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "sha256-q1cI7G/c7kVwMFMjbmdbNdrU6GHVmxTxcLbAfLaMEYU=";
+  };
+
+  cargoSha256 = "sha256-vmv7SOQqAPzYgKoTJKdiRrJ5t8iU1QlZy+VSlv8FRQE";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ SystemConfiguration ];
+
+  meta = with lib; {
```
Comment (alyssais) [inline pkgs/development/tools/riff/default.nix:27]:
```suggestion
  postInstall = ''
    wrapProgram $out/bin/riff --set-default RIFF_DISABLE_TELEMETRY 1
  '';

  meta = with lib; {
```

**Review by alyssais — COMMENTED** [review]
I think this would be the best way to solve the telemetry problem.  It avoids patching (which makes maintenance harder), and only changes the default, so people can still manually opt in by setting `RIFF_DISABLE_TELEMETRY=0`.

**Pull request closed without merging**